Drivers/Src/Lights.c

```diff
@@ -52,22 +76,25 @@ State Lights_Read(light_t light) {
  */
 void Lights_Set(light_t light, State state) {
     // Get internal lights from SPI
-    uint8_t txBuf[3] = {SPI_OPCODE_R, SPI_GPIOB, 0x00};
-    uint8_t rxBuf[2] = {0};
-    BSP_SPI_Write(txBuf, 3);
-
-    BSP_SPI_Read(rxBuf, 2);
+    uint8_t txReadBuf[2] = {SPI_OPCODE_R, SPI_GPIOB};
+    uint8_t txWriteBuf[3] = {SPI_OPCODE_W, SPI_GPIOB, 0x00};
+    uint8_t rxBuf[1] = {0};
+    
+    GPIO_WriteBit(GPIOA, GPIO_Pin_4, Bit_RESET);
+    BSP_SPI_Write(txReadBuf, 2);
+    BSP_SPI_Read(rxBuf, 1);
+    GPIO_WriteBit(GPIOA, GPIO_Pin_4, Bit_SET);
     
     uint8_t portc = BSP_GPIO_Read(LIGHTS_PORT);
```

Comment:
@IshDeshpa Should be uint16_t or else the left and right blink lights won't be affected because they are bits 8 and 9 in a 16 bit port

Reply:
Fixed